File: spring-flower/start/src/main/java/start/controller/statistics/Statistics.java
package start.controller.statistics;

import common.enums.OperationEnum;
import common.result.Result;
import model.vo.statistics.StatisticsVO;
import model.vo.statistics.OrderStatisticsVO;
import model.vo.statistics.TopStatisticsVO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import service.*;
import start.aop.OperationLogging;

import java.util.*;

@RestController
@RequestMapping("admin/statistics")
public class Statistics {

    @Autowired
    private FlowerOrderDetailService flowerOrderDetailService;

    private static final Integer TOP_NUMBER = 7;

    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/flower")
    public Result flowerSale() {
        List<StatisticsVO> statisticsVOList = flowerOrderDetailService.flowerSale();
        return Result.success(statisticsVOList);
    }

    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/festival")
    public Result festivalSale() {
        List<StatisticsVO> festivalStatisticsVOList = flowerOrderDetailService.festivalSale();
        return Result.success(festivalStatisticsVOList);
    }

    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/top1")
    public Result top1() {
        List<TopStatisticsVO> statisticsVOList = flowerOrderDetailService.top1();
        return Result.success(statisticsVOList);
    }

    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/top2")
    public Result top2() {
        List<TopStatisticsVO> festivalStatisticsVOList = flowerOrderDetailService.top2();
        return Result.success(festivalStatisticsVOList);
    }

    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/order")
    public Result order() {
        List<OrderStatisticsVO> orderStatisticsVOList = flowerOrderDetailService.order();
        return Result.success(orderStatisticsVOList);
    }

    // 今日统计：今日订单数、已支付订单数、今日营业额
    @OperationLogging(operation = OperationEnum.READ)
    @GetMapping("/today")
    public Result today() {
        List<OrderStatisticsVO> orderStatisticsVOList = flowerOrderDetailService.todayOrder();
        return Result.success(orderStatisticsVOList);
    }
}

File: vue-flower/src/api/admin/statistics.js
import request from '@/utils/admin/request.js'

/**
 * 营业数据统计 API（按 admin 接口文档对齐）
 * 后端控制器根路径：/admin/statistics，均为只读 GET 查询
 */

// 9.1 鲜花单品销量统计 - GET /admin/statistics/flower
// 响应 Result<List<StatisticsVO>>: { id, name, count, totalAccount }
export const getFlower = () => {
    return request({
        url: '/admin/statistics/flower',
        method: 'get'
    })
}

// 9.2 多花礼盒销量统计 - GET /admin/statistics/festival（按销量降序）
// 响应 Result<List<StatisticsVO>>
export const getFestival = () => {
    return request({
        url: '/admin/statistics/festival',
        method: 'get'
    })
}

// 9.3 鲜花热销榜 TopN - GET /admin/statistics/top1
// 响应 Result<List<TopStatisticsVO>>: { id, name, count }
export const getTop1 = () => {
    return request({
        url: '/admin/statistics/top1',
        method: 'get'
    })
}

// 9.4 多花礼盒热销榜 TopN - GET /admin/statistics/top2
// 响应 Result<List<TopStatisticsVO>>
export const getTop2 = () => {
    return request({
        url: '/admin/statistics/top2',
        method: 'get'
    })
}

// 9.5 订单状态分布统计 - GET /admin/statistics/order
// 响应 Result<List<OrderStatisticsVO>>: { status, name, count }
export const getOrder = () => {
    return request({
        url: '/admin/statistics/order',
        method: 'get'
    })
}

// 9.6 今日订单统计 - GET /admin/statistics/today
// 今日订单数、已支付订单数、今日营业额，响应 Result<List<OrderStatisticsVO>>
export const getTodayOrder = () => {
    return request({
        url: '/admin/statistics/today',
        method: 'get'
    })
}
